### DISCUSSION TITLE
switching to OpenSMTPD new config · poolp.org

### DISCUSSION THREAD

**Discussion by poolpOrg** [poolpOrg/poolp.org#94] [Articles]

**Comment by MrSorcus** [comment]
Can be ECC certs used with latest OpenSMTPD?

**Comment by poolpOrg** [comment]
no, there's a ticket opened.

**Comment by bg90jwef** [comment]
with the old grammar i used the following:

_accept from any for domain "example.com" recipient "foo@example.com" relay via tls+auth://exodus@smtp.provider.com auth <authinfo>_
_accept from any for any virtual { "@" => buggy } deliver to mbox_

to do some sort of catchall.
this worked well. any mail (local from system and remote from lan) was basically delivered to the local user _buggy_ which had a _.forward_ that contains foo@example.com. so it was relayed afterwards by the first rule.
any hint how this could be done in the new grammar? i have no idea how to implement the old-style => catchall.

**Comment by goekce** [comment]
a typo on the webpage https://poolp.org/posts/2018-05-21/switching-to-opensmtpd-new-config/

action act05 relay src <sources> helo-names <helonames>
should be
action act05 relay src <sources> helo-src <helonames>

**Comment by poolpOrg** [comment]
> Can be ECC certs used with latest OpenSMTPD?

This was implemented since my last response but only for OpenSMTPD linked against LibreSSL, the reason is that we don't just do ECC but we do ECC with private keys not exposed to the network-facing process and this relies on a low-level API which diverges between LibreSSL and OpenSSL.




> with the old grammar i used the following:
> 
> [...]
> 

Catch-all works the same, but the _virtual_ keyword is now associated to an action, not to a match.




> a typo on the webpage https://poolp.org/posts/2018-05-21/switching-to-opensmtpd-new-config/
> 
> action act05 relay src helo-names
> should be
> action act05 relay src helo-src

The article was written a while ago, I will fix the example and make it more clear :-)
